src/App.tsx

```diff
@@ -1,87 +1,96 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { remult } from "remult";
 import { Task } from "./shared/Task";
 import { TasksController } from "./shared/TasksController";
+import { observer } from "mobx-react-lite";
+import { action, makeAutoObservable, observable, runInAction } from 'mobx';
 
-const taskRepo = remult.repo(Task);
 
-async function fetchTasks(hideCompleted: boolean) {
-  return taskRepo.find({
-    limit: 20,
-    orderBy: { completed: "asc" },
-    where: { completed: hideCompleted ? false : undefined }
-  });
-}
 
 
+class Store {
+  taskRepo = remult.repo(Task);
+  readonly tasks = observable<Task>([]);
+  hideCompleted = false;
+  constructor() {
+    makeAutoObservable(this);
+  }
+  replaceTasks(t: Task[]) {
+    this.tasks.replace(t.map(t => makeAutoObservable({ ...t })));
+  }
+  async loadTasks() {
+    this.replaceTasks(await
+      this.taskRepo.find({
+        limit: 20,
+        orderBy: { completed: "asc" },
+        where: { completed: this.hideCompleted ? false : undefined }
+      }));
+  }
+  addTask() {
+    this.tasks.push(makeAutoObservable(new Task()))
+  }
+  async saveTask(task: Task) {
+    try {
+      const savedTask = await this.taskRepo.save(task);
+      runInAction(() => store.tasks[store.tasks.indexOf(task)] = savedTask);
+    } catch (error: any) {
+      alert(error.message);
+    }
+  }
+  async deleteTask(task: Task) {
+    await this.taskRepo.delete(task);
+    runInAction(() => store.tasks.remove(task));
+  }
+  async setAll(completed: boolean) {
+    await TasksController.setAll(completed);
+    this.loadTasks();
+  }
+}
+const store = new Store();
 
 function App() {
-  const [tasks, setTasks] = useState<Task[]>([]);
-  const [hideCompleted, setHideCompleted] = useState(false);
+  return <Todo store={store} />
 
+}
+const Todo: React.FC<{ store: Store }> = observer(({ store }) => {
   useEffect(() => {
-    fetchTasks(hideCompleted).then(setTasks);
-  }, [hideCompleted]);
-
-  const addTask = () => {
-    setTasks([...tasks, new Task()])
-  };
-  const setAll = async (completed: boolean) => {
-    await TasksController.setAll(completed);
-    setTasks(await fetchTasks(hideCompleted));
-  }
-
+    store.loadTasks()
+  }, [store.hideCompleted]);
```

Comment:
I'd tie this to the DOM's `DOMContentLoaded` event like this: `document.addEventListener('DOMContentLoaded', store.loadTasks, { once: true })` instead of React.
But I'm also painfully aware that that's not how most people write react apps nowadays. :(

Reply:
By using the use effect - I also get the automatic reload on change of hideCompleted - one way is to make the change through a mutation - I get that, but what's the "mobx" way of triggering an action on change of one or more state members?

Reply:
The mobx way of adding reactivity to react is via `mobx-react` bindings. The `mobx-react` package supports legacy (class based) components + `mobx-react-lite`, which supports only function components. I'd install only the latter and wrap each component in `observer`.

Reply:
Yeah, you can do that and in fact that's what the mobx-react bindings use. But why would you bother with a lower level and verbose API when you already have a solution: just wrap in `observer` and you're done.

Reply:
Just to make sure that I get you correctly.

Relace the code for loadTasks()
to be:
observer(()=>loadTasksImplementation())?